src/reanimated2/component/PerformanceMonitor.tsx

```diff
@@ -0,0 +1,203 @@
+      lastTime = time;
+      requestAnimationFrame(loop);
+    });
+  }
+
+  loop();
+}
+
+function getFps(renderTimeInMs: number): string {
+  'worklet';
+  return (1000 / renderTimeInMs).toFixed(1);
+}
+
+function getTimeDelta(
+  timestamp: number,
+  previousTimestamp: number | null
+): number {
+  'worklet';
+  return previousTimestamp !== null ? timestamp - previousTimestamp : 0;
+}
+
+function completeBufferRoutine(
+  buffer: CircularBuffer,
+  timestamp: number,
+  previousTimestamp: number,
+  totalRenderTime: SharedValue<number>
+) {
+  'worklet';
+  timestamp = Math.round(timestamp);
+  previousTimestamp = Math.round(previousTimestamp) ?? timestamp;
+
+  const droppedTimestamp = buffer.push(timestamp);
+  const nextToDrop = buffer.back()!;
+
+  const delta = getTimeDelta(timestamp, previousTimestamp);
+  const droppedDelta = getTimeDelta(nextToDrop, droppedTimestamp);
+
+  totalRenderTime.value += delta - droppedDelta;
+  return getFps(totalRenderTime.value / buffer.count);
+}
+
+function JsPerformance() {
+  const jsFps = useSharedValue<string | null>(null);
+  const totalRenderTime = useSharedValue(0);
+  const circularBuffer = createCircularDoublesBuffer(DEFAULT_BUFFER_SIZE);
+
+  useEffect(() => {
+    loopAnimationFrame((lastTime, time) => {
+      const currentFps = completeBufferRoutine(
+        circularBuffer,
+        time,
+        lastTime,
+        totalRenderTime
+      );
+
+      jsFps.value = currentFps;
+    });
+  });
+
+  const animatedProps = useAnimatedProps(() => {
+    const text = jsFps.value ?? 'N/A';
+    return { text, defaultValue: text };
+  });
+
+  return (
+    <View style={styles.container}>
+      <Text style={styles.headers}>JS FPS</Text>
+      <AnimatedTextInput
+        style={styles.text}
+        animatedProps={animatedProps}
+        editable={false}
+      />
+    </View>
+  );
+}
+
+function UiPerformance() {
+  const uiFps = useSharedValue<string | null>(null);
+  const totalRenderTime = useSharedValue(0);
+  const circularBuffer = useRef<CircularBuffer | null>(null);
+
+  useFrameCallback(({ timestamp }: FrameInfo) => {
+    if (circularBuffer.current === null) {
+      circularBuffer.current = createCircularDoublesBuffer(DEFAULT_BUFFER_SIZE);
+    }
+    const previousTimestamp = circularBuffer.current.front() ?? timestamp;
+    const currentFps = completeBufferRoutine(
+      circularBuffer.current,
+      timestamp,
+      previousTimestamp,
+      totalRenderTime
+    );
+    uiFps.value = currentFps;
+  });
+
+  const animatedProps = useAnimatedProps(() => {
+    const text = uiFps.value ?? 'N/A';
+    return { text, defaultValue: text };
+  });
+
+  return (
+    <View style={styles.container}>
+      <Text style={styles.headers}>UI FPS</Text>
+      <AnimatedTextInput
+        style={styles.text}
+        animatedProps={animatedProps}
+        editable={false}
+      />
+    </View>
+  );
+}
+
+export function PerformanceMonitor() {
+  return (
+    <View style={[styles.monitor]}>
```

Comment:
You don't need to add square brackets when using only one stylesheet style 😄 
```suggestion
    <View style={styles.monitor}>
```